### ISSUE TITLE
rustbuild: stdlib not installed for 2nd target triple

### ISSUE THREAD

**Issue by xi2** [rust-lang/rust#39235]
Hi. I am testing out rust 1.15.0 beta and I came across a problem installing from source:

### OS

Debian GNU/Linux 8 x86_64

### Rust version

10893a9 (1.15.0-beta.5)

### Actions

```
$ ./configure \
    --prefix=$HOME/rust \
    --enable-local-rust \
    --local-rust-root=$SNAPSHOT \
    --disable-rpath \
    --disable-manage-submodules \
    --target=x86_64-unknown-linux-gnu,i686-unknown-linux-gnu \
    --enable-vendor \
    --release-channel=beta
$ make
$ make install
```

### Expected outcome

`x86_64-unknown-linux-gnu` and `i686-unknown-linux-gnu` directories in `$HOME/rust/lib/rustlib`.

### Actual outcome

Only the `x86_64-unknown-linux-gnu` directory is present.

I am new to rust (just a couple of days) so forgive me if I've overlooked something. I'm pretty sure I installed 1.14.0 similarly and `i686-unknown-linux-gnu` was installed OK. That was using the old makefiles-based build system.

**Comment by alexcrichton** [comment]
IIRC this hasn't ever worked, but would be a great bug to fix with rustbuild!

**Comment by sezaru** [comment]
I've made a fix to this issue, you just need to apply this patch:
```
diff -ruN rustc-1.15.1-src/src/bootstrap/install.rs rustc-1.15.1-src_new/src/bootstrap/install.rs
--- rustc-1.15.1-src/src/bootstrap/install.rs	2017-02-09 01:37:48.000000000 +0000
+++ rustc-1.15.1-src_new/src/bootstrap/install.rs	2017-03-11 02:06:21.676817872 +0000
@@ -37,6 +37,12 @@
         install_sh(&build, "docs", "rust-docs", stage, host, prefix,
                    &docdir, &libdir, &mandir, &empty_dir);
     }
+    
+    for target in build.config.target.iter().filter(|target| *target != host) {
+        install_sh(&build, "std", "rust-std", stage, target, prefix,
+                   &docdir, &libdir, &mandir, &empty_dir);
+    }
+
     install_sh(&build, "std", "rust-std", stage, host, prefix,
                &docdir, &libdir, &mandir, &empty_dir);
     install_sh(&build, "rustc", "rustc", stage, host, prefix,
```
This will check the `build` structure for the targets provided on the `configure` stage, it will then remove any target that is equal to the host (since the host `rustlibs` will already be installed) and call the install function on it resulting in the installation of all the `rustlibs` for all the targets.

I don't know if this is the best approach or if there is some case that will fail and I'm not aware.

But, if it is a ok patch, can we apply it to the next version of rust since it seems that this is a major problem for a lot of users (at least for me it is since I'm trying to make the rust ebuild for gentoo support multiple targets and this bug is just what is preventing me).

Thanks

**Comment by alexcrichton** [comment]
@sezaru looks like a great start to me, you're more than welcome to send a PR!

I think that we'llalso want to update `step.rs` for the new dependencies, but other than that look great to me.

**Comment by Mark-Simulacrum** [comment]
I believe this is fixed now that #40479 landed, closing. Please let me know if I'm wrong though!
